packages/text-annotator-react/src/TextAnnotatorPopup/TextAnnotatorPopup.tsx

```diff
@@ -152,13 +139,12 @@ export const TextAnnotatorPopup = (props: TextAnnotationPopupProps) => {
             event
           })}
 
-          {/* It lets keyboard/sr users to know that the dialog closes when they focus out of it */}
-          <button className="popup-close-message" onClick={handleClose}>
-            This dialog closes when you leave it.
-          </button>
+          <span className="r6o-popup-sr-only" aria-live="assertive">
+            {props.ariaCloseWarning || 'This dialog will close when you leave it.'}
+          </span>
```

Comment:
Unfortunately, that message won't be picked by the screenreader. Because `aria-live` announces only the changes in the content within the live area. Therefore, the message will simply be missed ;(

Reply:
That's why I used the `@react-aria/live-announcer` package in https://github.com/recogito/text-annotator-js/pull/141. Furthermore, the live area won't even announce messages that are repeated! So I needed to trick it by inserting a random number of `&nbsp;` characters into it. 
[See the source](https://github.com/recogito/text-annotator-js/pull/141/files#diff-86cf4735dfec418174559a308bca544a80c6178258372614d602a699166b0602R27-R28)

Reply:
Ok, interesting. One thing I changed is that I made this a `<span>` rather than a button. Would a button make the message appear?

Reply:
And because of those reasons I went with the button, recommended by a11y expect. Because it's an explicit "speedbump" that will require an action from a user

Reply:
Or does this mean we should skip this altogether? (In favour of #141)

Reply:
Unfortunately, they are unrelated. In #141 I added a navigation message that will help users to know that they should press `Tab` to navigate to the popup. But it doesn't cover the focus-out speedbump. It was added in https://github.com/recogito/text-annotator-js/pull/118, [source](https://github.com/recogito/text-annotator-js/pull/118/files#diff-ad12777004d12b40f8adf25b1092fccc60c12ea2dd2d352ec0260079190780e8R155-R158)

Reply:
> Would a button make the message appear?

Yeah, it would, because the button is natively focusable. That's why the `r6o-popup-sr-only` has those `&:focus` appearing styles: https://github.com/recogito/text-annotator-js/blob/8d113dc93167df8d48521f331649202bf6cf32a1/packages/text-annotator-react/src/TextAnnotatorPopup/TextAnnotatorPopup.css#L20-L30

Reply:
Ok, sounds good. I'll revert to the button then. But I wonder if the default message should be modified accordingly. (It is still a button, so users with screen readers would also be able to 'click' it, right?) How about, simply 'Close this dialog'? Or 'Click to close this dialog. Leaving the dialog will close automatically.'?

Reply:
Or perhaps more compact: __Click or leave this dialog to close it.__

Reply:
> Or 'Click to close this dialog. Leaving the dialog will close automatically.'?

I like this option ✅ 
It's pretty clear and fits all the purposes